e2e/create-cluster_test.go

```diff
@@ -788,5 +791,60 @@ var _ = Describe("CreateCluster", func() {
 
 		By("waiting all tenant Pods to be Ready")
 		waitForTenantPods()
+
+		By("read the VMI again after it was recreated")
+		Eventually(func() *kubevirtv1.VirtualMachineInstance {
+			chosenVMI, err = virtClient.VirtualMachineInstance(namespace).Get(chosenVMI.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			return chosenVMI
+		}).
+			WithTimeout(time.Minute * 5).
+			WithPolling(time.Second * 10).ShouldNot(BeNil())
+
+		By("drain the node that hosts the vmi pod")
+		var nodeName string
+		// assuming we're not in migration, there should be one active pod
+		for _, nodeName = range chosenVMI.Status.ActivePods {
+			break
+		}
+
+		Expect(nodeName).ToNot(BeEmpty())
+		Expect(*chosenVMI.Spec.EvictionStrategy).Should(Equal(kubevirtv1.EvictionStrategyExternal))
+
+		var pods *corev1.PodList
+		Eventually(func() []corev1.Pod {
+			labelSelector := fmt.Sprintf("%s=%s", kubevirtv1.CreatedByLabel, chosenVMI.GetUID())
+			fieldSelector := fmt.Sprintf("status.phase==%s", corev1.PodRunning)
+			pods, err = virtClient.CoreV1().Pods(chosenVMI.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector})
+			Expect(err).ToNot(HaveOccurred())
+
+			return pods.Items
+		}, 90*time.Second, 500*time.Millisecond).Should(HaveLen(1))
+
+		By("evict the VMI pod")
+		Eventually(func() error {
+			return virtClient.CoreV1().Pods(namespace).EvictV1(context.Background(), &policyv1.Eviction{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pods.Items[0].Name,
+				},
+				DeleteOptions: &metav1.DeleteOptions{
+					GracePeriodSeconds: pointer.Int64(60 * 10), // 10 minutes
+				},
+			})
+		}).WithTimeout(time.Minute*5).
+			WithPolling(time.Second*10).
+			Should(Succeed(), "pod eviction was not done on time")
+
+		Eventually(func() types.UID {
+			vmiAfterEviction, err := virtClient.VirtualMachineInstance(namespace).Get(chosenVMI.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
there's goign to be a brief period of time where the old VMI is deleted, and the new VMI doesn't exist. So this might return an error during that time. 

Maybe just ignore the error if the error indicates the VMI doesn't exist?